x-pack/plugins/security_solution/public/management/components/endpoint_action_failure_message/endpoint_action_failure_message.tsx

```diff
@@ -24,51 +37,97 @@ export const EndpointActionFailureMessage = memo<EndpointActionFailureMessagePro
         return null;
       }
 
-      const errors: string[] = [];
+      const allAgentErrors: Array<{ name: string; errors: string[] }> = [];
 
       // Determine if each endpoint returned a response code and if so,
       // see if we have a localized message for it
-      if (action.outputs) {
+      // if there are multiple agents, we need to show the outputs/error message for each agent
+      if (action.outputs || (action.errors && action.errors.length)) {
```

Comment:
can you add tests for this component that validates the different conditions implemented.

(I actually thought we had tests for it, but maybe it was only the component that first implemented it 🤔 )

Reply:
c3f648de0079c2bc3df54d829909c46d1ce6298f